system/jlib/jdebug.cpp

```diff
@@ -1967,10 +1967,17 @@ class CExtendedStats  // Disk network and cpu stats
         memset(&oldnet, 0, sizeof(oldnet));
         memset(&newnet, 0, sizeof(newnet));
         ndisks = 0;
+        kbufmax = 0;
+#ifdef __linux__
         if (printklog)
-            kbufmax = 0x1000;
-        else
-            kbufmax = 0;
+        {
+            // kernel ring buffer size - supported since 2.6.6
+            // ignore kern msgs for anything older
+            int klogmax = klogctl(10, NULL, 0);
```

Comment:
It cannot grow larger, it is a kernel setting built into kernel so it will never change size unless a new kernel is loaded that was built with a different size.

SYSLOG_ACTION_SIZE_BUFFER is not defined, which is why thr 10 and the 3 are used.
The manpage states:
The symbolic names are defined in the kernel source, but are not exported to user space

There is an issue with getKLog() anyway in what it returns when an amount of data read less than previous - see linked issue HPCC-15703.  The getKLog() needs to be rewritten to locate the start of new data correctly.  And it would be good to read in only the last 4k or so anyway, instead of the entire buffer, to speed it up.

If we handle the last msg position properly so we can find all new data accurately then we could limit how much we read to a smaller amount for performance.

I will close this PR and rewrite getKLog() to resolve this and HPCC-15703